lib/facter/ec2.rb

```diff
@@ -46,8 +46,15 @@
     end
 
     setcode do
-      @querier.fetch
+      userdata = @querier.fetch
+      begin
+        userdata.encode(Encoding::UTF_8)
+      rescue EncodingError
```

Comment:
Yeah... you'd have to use iconv for 1.8.7, or just skip that?

Also watch out for ruby 2.1, as that was the version that added http auto-decompression, which is the issue PMT ran into.